helix-core/src/main/java/org/apache/helix/task/AbstractTaskDispatcher.java

```diff
@@ -306,9 +306,25 @@ public void updatePreviousAssignedTasksStatus(
           } else if (jobState == TaskState.IN_PROGRESS
               && (jobTgtState != TargetState.STOP && jobTgtState != TargetState.DELETE)) {
             // Job is in progress, implying that tasks are being re-tried, so set it to RUNNING
-            paMap.put(pId,
-                new JobRebalancer.PartitionAssignment(instance, TaskPartitionState.RUNNING.name()));
-            assignedPartitions.get(instance).add(pId);
+            String currentStateString =
+                currStateOutput.getCurrentState(jobResource, new Partition(pName), instance);
+            TaskPartitionState stateFromContext = jobCtx.getPartitionState(pId);
+
+            // If current state is null and context is INIT, the task should be dropped and
+            // reassigned in next pipeline.
+            if (currentStateString == null && stateFromContext != null) {
+              // Considered as Dropped case
```

Comment:
Why would this case be considered as dropped case? What does dropped case even mean?